include/GoogleSync/GoogleSync.php

```diff
@@ -384,14 +472,13 @@ public function getGoogleEventById($event_id)
     {
 
         // Make sure the calendar service is set up
-        if (!isset($this->gService)) {
-            $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'The Google Service is not set up. See setGService Method.');
+        if (!$this->isServiceExists()) {
             return false;
         }
 
         if (empty($event_id)) {
-         // If we didn't get passed an event, return null
-            return;
+            // If we didn't get passed an event, return null
+            throw new \InvalidArgumentException('event ID is empty');
```

Comment:
This can be null. Can't throw an exception here.

Reply:
If it is the case please think about it: What does it means if this methos returns `null`? I mean this method also returns `null` when did not find any google event and also returns null if given parameter $event_id was `null`. So how the caller can understend what is going on? If a caller can not decide what was the reason of failure, then calles has not any chance to handle the error properly. At least add a log note to give a chence to the end-user to figurate out what just happend here..

Reply:
In other way, the problem here is: This method getGoogleEventById() called at doSync() and passing the return (which could be `null`) to pushPullSkip() method. this method try to resolve the gevent, but it returns a `false` as it should, so the $dowhat could be `false` at this point:
```
            $gevent = $this->getGoogleEventById($meeting->gsync_id);
            $dowhat = $this->pushPullSkip($meeting, $gevent);
```
and later in a dafeult case of the following switch, it will be concatenated into a log not:
```
$this->logger->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'pushPullSkip() returned unknown value: ' . $dowhat . ' for event: ' . $meeting->name);
```
but the `false` value converted to an empty string, which is very confusing to a human reader just like the end-users or custom developers who will calling this method.

Reply:
I'll find a better way to handle this. I just had to set it back to returning null because it was breaking and I was working on another part of the code.
Thank you for the explanations.

Reply:
Ok. This is how I'm handling this. Let me know what you think.

First, I'm checking in doSync if $meeting->gsync_id is empty. If it is, I'm not calling getGoogleEventById at all and declaring $gevent as null then skipping right to pushPullSkip.

Second, I'm changing getGoogleEventById so that it only ever returns the google event or null. Never false. If the google calendar service isn't set up, it throws an exception. It should NEVER get there.

I've also split out the large case statements from doSync into doAction, and reduced it to a single statement.

Reply:
Thank you, it sounds right, will see.
Also I wrote a small gist example to explan why and how to use exception to solve this kind of problems:
https://gist.github.com/gymad/e8ed05aaf6e7e9c05472c99d1007c5a3

Reply:
Just in time. I'm working on other parts that could use it. Thanks!